sdk/identity/Azure.Identity/src/EnvironmentCredential.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Identity
+{
+    public class EnvironmentCredential : TokenCredential
+    {
+        private TokenCredential _credential = null;
+
+        public EnvironmentCredential()
+            : this(null)
+        {
+        }
+
+        public EnvironmentCredential(IdentityClientOptions options)
+        {
+            string tenantId = Environment.GetEnvironmentVariable("AZURE_TENANT_ID");
+            string clientId = Environment.GetEnvironmentVariable("AZURE_CLIENT_ID");
+            string clientSecret = Environment.GetEnvironmentVariable("AZURE_CLIENT_SECRET");
```

Comment:
Is the intention to also handle certificates? If so, was that omitted for brevity?

Reply:
Yes I'll be adding support for that shortly.